### PR TITLE
adds /api endpoint to complete task 3

### PR DESCRIPTION
**Pull request by msmi1433**

### CODE REVIEW

File: __tests__/getEndpoints.test.js [hunk 1]
```diff
@@ -25,4 +25,15 @@ describe("GET endpoints", () => {
         });
     });
   });
+  describe("CORE: GET /api", () => {
+    test("200: responds with object that matches local enspoints JSON", () => {
+      const endpointsJson = require("../endpoints.json");
```
Comment (Dutton1990) [inline __tests__/getEndpoints.test.js:30]: Nice thinking!

File: app.js [hunk 2]
```diff
@@ -1,7 +1,16 @@
 const express = require("express");
 const app = express();
-const { getTopics } = require("./controllers");
+const { getTopics, getEndpoints } = require("./controllers");
+
+app.use(express.json());
```
Comment (Dutton1990) [inline app.js]: You don't need this yet, in future only add code relevant to the ticket

File: controllers/endpoints.controller.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+const { selectAllEndpoints } = require("../models/index");
+
+exports.getEndpoints = (req, res) => {
+  selectAllEndpoints().then((endpoints) => {
+    res.status(200).send(endpoints);
```
Comment (Dutton1990) [inline controllers/endpoints.controller.js]: send your endpoints in an object so you can give it a key that makes more sense than just 'body'

File: models/endpoints.model.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+const fs = require("fs/promises");
+
+exports.selectAllEndpoints = () => {
+  return fs.readFile(`${__dirname}/../endpoints.json`, "utf-8").then((file) => {
+    return JSON.parse(file);
+  });
+};
```
Comment (Dutton1990) [inline models/endpoints.model.js]: You don't really need a model for this one, models should just deal with the database. You could do this in the controller if you wanted to.

**Review by Dutton1990 — APPROVED** [review]
A few suggestions Myles but you can merge and continue when you're ready nice job!

**Comment by msmi1433** [comment]
> A few suggestions Myles but you can merge and continue when you're ready nice job!

Thanks Christian :)

**Pull request merged**